src/rpc/blockchain.cpp

```diff
@@ -1122,15 +1122,20 @@ UniValue getblockchaininfo(const UniValue &params, bool fHelp)
 
     if (fPruneMode)
     {
-        CBlockIndex *block = chainActive.Tip();
+        if (!fPruneWithMask)
         {
-            READLOCK(cs_mapBlockIndex);
+            CBlockIndex *block = chainActive.Tip();
             while (block && block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA))
```

Comment:
why take out the READLOCK()? We need that for checking nStatus.

Reply:
good catch, i will add that back in.